docs/source/server.rst

```diff
@@ -0,0 +1,130 @@
+  on the server, this its unique identifier. See Source, POST for how these IDs are generated.
+
+Returns
+~~~~~~~
+
+- ``version``, string: the server's Intake version
+
+- ``sources``, list of objects: the main payload, where each object contains a ``name``, and the result of calling
+  ``.describe()`` on the corresponding data-source, i.e., the container type, description, metadata.
+
+- ``metadata``, object: any metadata associated with the whole catalog
+
+Source handler, GET
+-------------------
+
+Fetch information about a specific source. This is the random-access variant of the Info, GET route, by which
+a particular data-source can be accessed without paginating through all of the sources.
+
+Parameters
+~~~~~~~~~~
+
+- ``name``, string: the data source name being accessed, one of the members of the catalog
+
+- ``source_id``, uuid string: when the catalog being accessed is not the root catalog, but an open data-source
+  on the server, this its unique identifier. See Source, POST for how these IDs are generated.
+
+Returns
+~~~~~~~
+
+Same as one of the entries in ``sources`` for Info, GET: the result of ``.describe()`` on the given data-source in the
+server
+
+Source handler, POST, action="open"
+-----------------------------------
+
+This is a more involved processing of a data-source, and, if successful, returns one of two possible scenarios:
+
+- direct-access, in which all the details required for reading the data directly from the client are passed, and
+  the client then creates a local copy of the data source and needs no further involvement from the server in order
+  to fetch the data
+
+- remote-access, in which the client is unable or unwilling to create a local version of the data-source, and instead
+  created a remote data-source which will fetch the data for each partition from the server.
+
+The set of parameters supplied and the server/client policies will define which method of access is employed. In the
+case of remote-access, the data source is instantiated on the server, and ``.discover()`` run on it. The resulting
+information is passed back, and must be enough to instantiate a subclass of ``intake.container.base.RemoteSource``
+appropriate for the container of the data-set in question (e.g., ``RemoteArray`` when ``container="ndarray"``).
+In this case, the response also includes a UUID string for the open instance on the server, referencing the
+cache of open sources maintained by the server.
+
+Note that "opening" a data entry which is itself is a catalog implies instantiating that catalog object on the
+server and returning its UUID, such that a listing can be made using Info, GET or Source, GET.
+
+Parameters
+~~~~~~~~~~
+
+- ``name``, string: the data source name being accessed, one of the members of the catalog
+
+- ``source_id``, uuid string: when the catalog being accessed is not the root catalog, but an open data-source
+  on the server, this its unique identifier.
```

Comment:
Might be worth adding, just to be extra clear: "If the catalog being accessed is the root Catalog, this parameter should be omitted."